pkg/controller/atlasdeployment/atlasdeployment_controller_test.go

```diff
@@ -851,8 +854,40 @@ func TestReconciliation(t *testing.T) {
 			WithStatusSubresource(bPolicy, bSchedule).
 			Build()
 
+		searchAPI := atlasmock.NewAtlasSearchApiMock(t)
+		searchAPI.EXPECT().GetAtlasSearchDeployment(context.Background(), project.ID(), deployment.Spec.DeploymentSpec.Name).
+			Return(admin.GetAtlasSearchDeploymentApiRequest{ApiService: searchAPI})
+		searchAPI.EXPECT().GetAtlasSearchDeploymentExecute(mock.Anything).
+			Return(
+				&admin.ApiSearchDeploymentResponse{
+					GroupId:   pointer.MakePtr(project.ID()),
+					StateName: pointer.MakePtr("IDLE"),
+					Specs: &[]admin.ApiSearchDeploymentSpec{
+						{
+							InstanceSize: "S100_LOWCPU_NVME",
+							NodeCount:    4,
+						},
+					},
+				},
+				&http.Response{},
+				nil,
```

Comment:
are we testing all the cases with this one unit test? what is the coverage?

Reply:
I am still missing coverage of the main reconciliation method in unit tests. See comments below.